cmake/bazel.cmake

```diff
@@ -22,15 +22,25 @@
 #
 ########################################################################
 
-if(NOT CMAKE_CROSSCOMPILING)
-    find_package(CUDA QUIET)
-endif(NOT CMAKE_CROSSCOMPILING)
-
-option(WITH_GPU     "Compile Source Code with NVIDIA GPU"     ${CUDA_FOUND})
+option(WITH_GPU     "Compile Source Code with NVIDIA GPU"     OFF)
 option(WITH_TESTING "Compile Source Code with Unit Testing"   ON)
 option(WITH_PYTHON  "Compile Source Code with Python"         ON)  
 option(WITH_MSVC_MT "Compile Source Code with MSVC /MT"       OFF)
 
+if(NOT CMAKE_CROSSCOMPILING)
+    find_package(CUDA REQUIRED)
+endif(NOT CMAKE_CROSSCOMPILING)
+
+if(CUDA_FOUND)
+  set(WITH_GPU ON)
+endif(CUDA_FOUND)
```

Comment:
This if-endif might not be very right as it prevents people from building a non-GPU version on a computer with CUDA installed. Not sure what would be the best solution. @gangliao

Reply:
I think we should still keep the logic order of original order. It's very useful if developers want to build a CPU-version on a computer with CUDA installed. @wangkuiyi 

```
cmake -DWITH_GPU=OFF ..
```

Reply:
Also, keyword `REQUIRED` means CUDA must be founded. `find_package(CUDA REQUIRED)`  will raise the compilation error.